build/templates/_attributes.py.mako

```diff
@@ -3,10 +3,15 @@ ${template_parameters['encoding_tag']}
 <%
     module_name = template_parameters['metadata'].config['module_name']
     config = template_parameters['metadata'].config
+    datetime_library = config['datetime_library']
 %>\
 import ${module_name}._converters as _converters
 
+% if config['uses_hightime']:
+import hightime
+% else:
 import datetime
+% endif
```

Comment:
So we either use `hightime` all the time or `datetime` all the time?
I think it would be better to use `datetime` all the time __unless__ hightime is needed. This means we always import `datetime` and optionally import `hightime`.

Reply:
Yes, it's one or the other. Given `hightime` is designed to be type-compatible with `datetime`, I thought it would be better for usability to have all `timedelta` objects to be the same type.

As an added bonus, development and maintenance becomes easier and less error-prone. If we use both types, attribute classes:
- `AttributeViInt32TimeDeltaMilliseconds`
- `AttributeViReal64TimeDeltaSeconds`
and converters:
- `convert_seconds_real64_to_timedelta`
- `convert_seconds_real64_to_timedeltas`
- `convert_timedelta_to_seconds_real64`
- `convert_timedelta_to_milliseconds_int32`
- `convert_timedeltas_to_seconds_real64`
will need to be duplicated. Metadata will have to carefully maintained to use the correct attribute class or converter.

Can you please elaborate how targeted use of `hightime` would be better?

Reply:
> Can you please elaborate how targeted use of hightime would be better?
 
1. Similar methods across different nimi-python modules would return precisely the same type.
2. When we fix #744 adn #1382 we won't change the type returned by some existing functions that are precise enough with `datetime.timedelta`, duck typing notwithstanding.
3. When other driver runtimes add functions that require more precision, we won't be impacting existing functions like in no. 2.

I welcome the opinion of @texasaggie97 and @thejcannon

Reply:
> As an added bonus, development and maintenance becomes easier and less error-prone.

I do see this benefit.

Reply:
Hmm, I'm sold on points 2 and 3. I'll start looking into making the required changes. Should be relatively straight-forward.

Reply:
I don't think I really have an opinion here. 

`hightime` sometimes works for me:
If we do `hightime` sometimes, we can always switch to `hightime` always if we want. It's a lot harder if we do `hightime` always and want to switch it to do `hightime` sometimes.

Reply:
I think consistency within the driver > consistency across drivers
Also, since the normal `datetime` and `timedelta` are base classes for hightime, something like `dt 
= hightime.datetime(...); isinstance(dt, datetime.datetime)` will return true.

Overall, it seems like a lot of overhead for something that won't make a material difference.

Reply:
TL;DR: After offline discussion, we decided to replace `datetime.timedelta` with `hightime.timedelta` across the board in all usages of `timedelta` in all modules in nimi-python.

Pros:
- Code generator is simpler, no `if-else` cases
- No need to carefully curate metadata, with respect to which `timedelta` to use
- Consistency within and across nimi-python modules
- No churn if underlying driver runtime or a new device family starts supporting higher resolution for some property or method parameter

Cons:
- Currently, certain modules in nimi-python do not have a need for higher resolution provided by `hightime`. Users of those modules will be forced to install a third-party library - `hightime` as a dependency and use `hightime.timedelta` type.

Given that `datetime.datetime` and `datetime.timedelta` are base classes for the respective `hightime` variants, with duck-typing, users should not have to do anything different to use `hightime` types, other than importing from a different package.

Pros clearly outweigh the cons, thus the decision to move to `hightime.timedelta` wholesale.